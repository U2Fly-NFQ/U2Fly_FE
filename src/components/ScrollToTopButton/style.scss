.scroll-to-top-container {
  position: fixed;
  bottom: 40px;
  right: 25px;
  z-index: 20;
}
