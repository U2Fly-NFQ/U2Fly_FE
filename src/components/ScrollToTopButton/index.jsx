import { Button } from 'antd'
import React, { useState, useEffect } from 'react'
import { motion, AnimatePresence } from 'framer-motion'

import './style.scss'

const ScrollToTopButton = () => {
  const [scrollPosition, setScrollPosition] = useState(0)

  useEffect(() => {
    const updatePosition = () => {
      setScrollPosition(window.scrollY)
    }

    window.addEventListener('scroll', updatePosition)

    return () => window.removeEventListener('scroll', updatePosition)
  }, [])

  const goToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth',
    })
  }

  return (
    <AnimatePresence>
      {scrollPosition > 100 && (
        <motion.div
          className="scroll-to-top-container"
          initial={{ y: -100, opacity: 0 }}
          animate={{ y: 0, opacity: 1, transition: { duration: 0.6 } }}
          exit={{ y: -100, opacity: 0, transition: { duration: 0.6 } }}
          whileHover={{
            scale: 1.3,
            transition: { duration: 0.2 },
          }}
          whileTap={{ scale: 1 }}
        >
          <Button size="large" type="primary" shape="circle" onClick={goToTop}>
            <i className="fa-solid fa-arrow-up"></i>
          </Button>
        </motion.div>
      )}
    </AnimatePresence>
  )
}
export default ScrollToTopButton
