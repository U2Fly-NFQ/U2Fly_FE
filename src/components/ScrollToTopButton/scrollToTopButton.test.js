import renderer from 'react-test-renderer'
import React from 'react'
import ScrollToTopButton from './'

describe('Scroll to to top button test', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ScrollToTopButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
