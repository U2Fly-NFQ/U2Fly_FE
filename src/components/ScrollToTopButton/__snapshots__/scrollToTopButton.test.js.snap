// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll to to top button test renders correctly 1`] = `null`;
